hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/strategy/CompactionStrategy.java

```diff
@@ -102,4 +105,19 @@ public List<HoodieCompactionOperation> orderAndFilter(HoodieWriteConfig writeCon
   public List<String> filterPartitionPaths(HoodieWriteConfig writeConfig, List<String> allPartitionPaths) {
     return allPartitionPaths;
   }
+
+  /**
+   * Creates a new instance of the provided class.
+   * @param strategyClass class name of the strategy
+   * @param hoodieEngineContext an engine context which may be used by the compaction strategy
+   * @param hoodieTable a hoodie table which may be used by the compaction strategy
+   * @return a new instance of the provided class
+   */
+  public static CompactionStrategy create(String strategyClass, HoodieEngineContext hoodieEngineContext, HoodieTable hoodieTable) {
```

Comment:
@the-other-tim-brown HoodieTable interface exposes different write APIs (insertXXX, bulkInsertXXX, upsertXXX, deleteXXX) which should not be called by the CompactionStrategy implementation. 

Can you define an interface to expose the subset of APIs that you want to compaction strategy implementation to make use of ?